sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1158,12 +1165,77 @@ def test_large_regularization():
     # regularization parameters
     for penalty in ['l2', 'l1', 'elasticnet']:
         model = SGDClassifier(alpha=1e5, learning_rate='constant', eta0=0.1,
-                              n_iter=5, penalty=penalty, shuffle=False)
+                              penalty=penalty, shuffle=False,
+                              tol=-np.inf, max_iter=6)
         with np.errstate(all='raise'):
             model.fit(iris.data, iris.target)
         assert_array_almost_equal(model.coef_, np.zeros_like(model.coef_))
 
 
+def test_tol_parameter():
+    # Test that the tol parameter behaves as expected
+    X = StandardScaler().fit_transform(iris.data)
+    y = iris.target
+
+    # With tol == -np.inf, the number of iteration should be equal to max_iter
+    max_iter = 42
+    model_0 = SGDClassifier(tol=-np.inf, max_iter=max_iter).fit(X, y)
+    assert_equal(max_iter, model_0.n_iter_)
+
+    # With tol > -np.inf, the number of iteration should be less than max_iter
+    max_iter = 2000
+    model_1 = SGDClassifier(tol=0, max_iter=max_iter).fit(X, y)
+    assert_greater(max_iter, model_1.n_iter_)
```

Comment:
Can you please fix the `random_state` of `SGDClassifier` and max sure that `assert_greater(model_1.n_iter_, 5)` for instance?